examples/third_party/mesa/BUILD.mesa.bazel

```diff
@@ -38,7 +38,21 @@ meson_with_requirements(
             "PATH": "$$(dirname $$EXT_BUILD_ROOT$$/$(location @bison//:gen_dir))/bison/bin:$$(dirname $$EXT_BUILD_ROOT$$/$(location @flex//:flex_exe)):$$EXT_BUILD_DEPS$$/bin/m4/bin/:$$(dirname $$EXT_BUILD_ROOT$$/$(PYTHON3)):$$PATH",
         },
     }),
+    copts = select({
+        # "@openssl//:msvc_compiler": ["/std:c++14"],
+        "@platforms//os:linux": ["-std=c++14"],
+        "//conditions:default": [],
+    }),
+
     lib_source = ":all_srcs",
+    linkopts = select({
+        "@openssl//:msvc_compiler": [
+            "ws2_32.lib",
+        ],
+        "@platforms//os:linux": ["-lstdc++"],
```

Comment:
what if you're using libc++ via clang? this arg will break a libc++ build with clang.

Reply:
by setting `CXX` and `CC` both compilers become `/usr/bin/gcc`, which is then used to compile cpp files in CI.
during [linking with cpp file](https://buildkite.com/bazel/rules-foreign-cc/builds/5111#018ae1da-444c-4dab-8efa-0b0d2612129d/6-3800), it hits 
```
undefined reference to `operator new
```
because of missing C++ support runtime library

Reply:
added additional logic to not use `lstdc++`